src/plugins/graph/models/graph-model.ts

```diff
@@ -178,11 +178,19 @@ export const GraphModel = TileContentModel
         return;
       }
       if (getAppConfig(self)?.getSetting("emptyPlotIsNumeric", "graph")) {
-        // If our graph doesn't have useful axes then set them
-        if (!self.getAttributeID("x") && !self.getAttributeID("y")) {
-          self.setAttributeID("x", self.data.attributes[0].id);
-          self.setAttributeID("y", self.data.attributes[1].id);
-        }
+        // We need attributes in the data model before we can set the x and y attributes of the graph
+        // If the source is a Dataflow tile with no recording, then SharedModel is not there yet
+        // and self.data.attributes exists but has no length. This causes an error in the mobx reaction.
+        // So in that case  we need to bail to avoid the mobx error
+        const attributesInDataSet = self.data.attributes.length > 0;
+        if (!attributesInDataSet) return;
+
+        // attribute ids are novel on new recordings,
+        // so we set new attributes on the condition that id in graph does not match id in data
+        const newX = self.getAttributeID("x") !== self.data.attributes[0].id;
+        const newY = self.getAttributeID("y") !== self.data.attributes[1].id;
+        if (newX) self.setAttributeID("x", self.data.attributes[0].id);
+        if (newY) self.setAttributeID("y", self.data.attributes[1].id);
```

Comment:
This appears to be where the reset we just identified as bugs to be fixed is happening. You should narrow down the circumstances under which this occurs, e.g. only when the current attribute assigned to the axis doesn't exist in the data.

Reply:
Yes, thanks. I was wondering if that might be the case. Will post back in a bit.